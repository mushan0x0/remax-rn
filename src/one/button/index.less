.disabled {
  //background-color: #dddddd;
  //color: #cccc;
}

.btn {
  background: #2059e1;
  border-radius: 10px;
  height: 94px;
  font-size: 36px;
  color: #fff;
  padding: 0 14px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
