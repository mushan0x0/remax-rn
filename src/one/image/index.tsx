import React, { useEffect, useState } from 'react';
import { Image, TouchableOpacity } from 'react-native';
// import FastImage from 'react-native-fast-image';
import transformStyles from '../../utils/transformStyles';
import NeedWrap from '../../utils/NeedWrap';
import { SvgXml } from 'react-native-svg';
// @ts-ignore
import resolveAssetSource from 'react-native/Libraries/Image/resolveAssetSource';

export default ({ src, className, style = {}, mode, onTap }: any) => {
  style = style instanceof Array ? Object.assign({}, ...style) : style;
  if (style) {
    Object.keys(style).forEach((key) => {
      if (style[key] === undefined) {
        delete style[key];
      }
    });
  }
  style = {
    ...className,
    ...style,
  };
  const [WH, setWH] = useState(style as any);
  const isStr = typeof src === 'string';
  useEffect(() => {
    if (src && mode === 'widthFix') {
      if (isStr) {
        Image.getSize(
          src,
          (width, height) => {
            setWH({ width, height });
          },
          () => {}
        );
      } else {
        setWH(resolveAssetSource(src));
      }
    }
  }, [isStr, mode, src]);
  const { width } = style;
  const multiple = (width || 0) / (WH.width || 0);
  return (
    <NeedWrap
      need={!!onTap}
      wrap={TouchableOpacity}
      wrapProps={{ onPress: onTap }}
    >
      {/svg/.test(src) ? (
        <SvgXml
          {...transformStyles(style as any)}
          xml={src.replace(/ (width|height)="\d+px"/g, ` $1="${width}px"`)}
        />
      ) : (
        <Image
          source={isStr ? { uri: src } : src}
          style={transformStyles({
            height: WH.height ? WH.height * multiple : undefined,
            ...style,
          } as any)}
          resizeMode={mode === 'aspectFit' ? 'contain' : undefined}
        />
      )}
    </NeedWrap>
  );
};
