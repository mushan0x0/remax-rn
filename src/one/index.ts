export * from './utils';
export { default as View } from './view';
export { default as Text } from './text';
export { default as Button } from './button';
export { default as Input } from './input';
export { default as Image } from './image';
export { default as Textarea } from './textarea';
