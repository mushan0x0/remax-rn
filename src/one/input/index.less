.input {
  //border: 1px solid #000;
  background-color: #fff;
}
