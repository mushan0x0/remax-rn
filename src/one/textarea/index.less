.input {
  border: 1px solid #000;
  background-color: #fff;
  padding-top: 0;
}
