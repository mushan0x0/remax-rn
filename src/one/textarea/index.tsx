import { TextInput } from 'react-native';
import React from 'react';
import styles from './index.less';

export default ({ style, onInput, onChange, ...props }: any) => {
  return (
    <TextInput
      textAlignVertical={'top'}
      style={{ ...styles.input, ...style }}
      {...props}
      onChangeText={(text) => {
        onInput?.({ target: { value: text } });
        onChange?.(text);
      }}
      underlineColorAndroid="transparent"
      multiline={true}
    />
  );
};
