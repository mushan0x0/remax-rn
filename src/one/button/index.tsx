import React from 'react';
import { View } from 'remax/one';
import styles from './index.less';

export default ({ onTap, children, disabled, style }: any) => {
  return (
    <View
      onTap={disabled ? undefined : onTap}
      className={styles.btn}
      style={{
        ...style,
        ...(disabled && styles.disabled),
      }}
    >
      {children}
    </View>
  );
};
