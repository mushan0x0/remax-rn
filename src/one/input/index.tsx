import { TextInput } from 'react-native';
import React, { forwardRef, useContext } from 'react';
import { View } from 'remax/one';
import transformStyles from '../../utils/transformStyles';
import { rpxToPx } from '@kqinfo/ui';
import { extendStyle } from '../view';

const Input = ({ style, ...props }: any) => {
  const {
    fontSize = rpxToPx(26),
    textAlign,
    ...parentStyle
  } = useContext(extendStyle);
  Object.keys(style).forEach((key) => {
    if (/em/i.test(style[key])) {
      const number = style[key].replace(/em/i, '');
      style[key] =
        +number *
        ((+typeof style.fontSize ? style.fontSize : fontSize) || rpxToPx(26));
    }
  });
  return <TextInput style={{ textAlign, ...style }} {...props} />;
};

export default forwardRef(
  (
    {
      style,
      className,
      placeholderStyle,
      onInput,
      onChange,
      onConfirm,
      type,
      ...props
    }: any,
    ref,
  ) => {
    if (type === 'number') {
      type = 'number-pad';
    } else {
      type = 'default';
    }
    const viewStyle = { ...className, ...style };
    return (
      <View style={viewStyle}>
        <Input
          style={transformStyles({
            backgroundColor: '#fff',
            paddingLeft: 0,
            paddingRight: 0,
            paddingTop: 0,
            paddingBottom: 0,
            fontSize: viewStyle.fontSize,
            ...className,
            ...style,
          })}
          underlineColorAndroid="transparent"
          placeholderTextColor={
            placeholderStyle?.color || style?.color || '#999'
          }
          onChangeText={(text: any) => {
            onInput?.({ target: { value: text } });
            onChange?.(text);
          }}
          onSubmitEditing={onConfirm}
          keyboardType={type}
          ref={ref}
          returnKeyType={'done'}
          {...props}
        />
      </View>
    );
  },
);
