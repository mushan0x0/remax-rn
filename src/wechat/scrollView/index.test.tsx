import TestRenderer from 'react-test-renderer';
import Scroll from './index';
import { View } from '@/one';
import React from 'react';

jest.useFakeTimers();

describe('Scroll', () => {
  it('基本渲染', () => {
    expect(
      TestRenderer.create(
        <Scroll
          style={{ flexWrap: 'no-wrap' }}
          className={{ background: 'red' }}
          scrollX
        >
          <View>233</View>
          <View>233</View>
        </Scroll>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
