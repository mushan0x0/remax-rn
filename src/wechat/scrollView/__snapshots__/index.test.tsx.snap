// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll 基本渲染 1`] = `
<RCTScrollView
  horizontal={true}
  keyboardDismissMode="on-drag"
  keyboardShouldPersistTaps={true}
  onScroll={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "display": "flex",
      "flexWrap": "no-wrap",
      "transform": Array [
        Object {
          "rotate": "30deg",
        },
      ],
    }
  }
>
  <View>
    <View
      activeOpacity={0.5}
      className="view"
      onPress={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "opacity": undefined,
          "position": undefined,
        }
      }
    >
      <Text
        className="view"
        style={
          Object {
            "color": undefined,
            "fontSize": 26,
            "fontWeight": undefined,
            "height": undefined,
            "lineHeight": undefined,
            "paddingTop": undefined,
            "textAlign": undefined,
            "textAlignVertical": "center",
          }
        }
      >
        233
      </Text>
    </View>
    <View
      activeOpacity={0.5}
      className="view"
      onPress={[Function]}
      style={
        Object {
          "borderRadius": undefined,
          "display": "flex",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "opacity": undefined,
          "position": undefined,
        }
      }
    >
      <Text
        className="view"
        style={
          Object {
            "color": undefined,
            "fontSize": 26,
            "fontWeight": undefined,
            "height": undefined,
            "lineHeight": undefined,
            "paddingTop": undefined,
            "textAlign": undefined,
            "textAlignVertical": "center",
          }
        }
      >
        233
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
